internal/powerplatform/services/data_record/api_data_record.go

```diff
@@ -181,41 +252,81 @@ func (client *DataRecordClient) GetRelationData(ctx context.Context, recordId, e
 	return value, nil
 }
 
-func (client *DataRecordClient) GetEntityRelationDefinitionInfo(ctx context.Context, environmentId string, entityLogicalName string, relationLogicalName string) (tableName string, primaryIdFieldName string, err error) {
+func (client *DataRecordClient) GetTableSingularNameFromPlural(ctx context.Context, environmentId, logicalCollectionName string) (*string, error) {
```

Comment:
if I remember correctly this method gets plural table name from singular

Reply:
that got refactored